core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1731,6 +1764,46 @@ object Parser {
     */
   def end: Parser0[Unit] = Impl.EndParser
 
+  /** return the position the parser is at
+    */
+  def careting: Parser0[Caret] = Impl.WithCaret
+
+  /** return the result of the parser together
+    * with the position of the starting of the consumption
+    */
+  def careted0[A](pa: Parser0[A]): Parser0[(A, Caret)] =
+    (careting ~ pa).map { case (l, x) => (x, l) }
+
+  /** return the result of the parser together
+    * with the position of the starting of the consumption
+    */
+  def careted[A](pa: Parser[A]): Parser[(A, Caret)] =
+    (careting.with1 ~ pa).map { case (l, x) => (x, l) }
+
+  /** replace the result of the parser with with the
+    * span of position of the string it consumed
+    */
+  def spanning0[A](pa: Parser0[A]): Parser0[Span] =
```

Comment:
can we change this to 'def spanOf`?

Reply:
I think `span` is better given we have `string`?

Reply:
Sure, that's fine.